src/cogent3/app/data_store_new.py

```diff
@@ -186,66 +188,57 @@ def __init__(
         self._not_completed = []
         self._limit = limit
         self._verbose = verbose
-        self._checksums = {}
         self._md5 = md5
         self._source_check_create(if_dest_exists)
 
     def _source_check_create(self, if_dest_exists):
         if not is_master_process():
             return
+
         sub_dirs = [_NOT_COMPLETED_TABLE, _LOG_TABLE, _MD5_TABLE]
-        if if_dest_exists is READONLY and not self.source.exists():
-            raise IOError("must exist")
-        elif if_dest_exists is RAISE and self.source.exists():
-            raise IOError("must not exist")
+        source = self.source
+        if if_dest_exists is READONLY and not source.exists():
+            raise IOError(f"'{source}' does not exist")
+        elif if_dest_exists is RAISE and source.exists():
+            raise IOError(f"'{source}' exists")
         elif if_dest_exists is READONLY:
             return
 
-        path = Path(self.source)
-        if path.exists() and if_dest_exists == RAISE:
-            raise FileExistsError(f"'{self.source}' exists")
-        elif path.exists() and if_dest_exists == OVERWRITE:
+        if source.exists() and if_dest_exists is OVERWRITE:
             for sub_dir in sub_dirs:
-                try:
-                    shutil.rmtree(self.source / sub_dir)
-                except:
-                    pass
-        elif not path.exists():
-            path.mkdir(parents=True, exist_ok=True)
+                with contextlib.suppress((FileNotFoundError, NotADirectoryError)):
+                    shutil.rmtree(source / sub_dir)
+        elif not source.exists():
+            source.mkdir(parents=True, exist_ok=True)
 
         for sub_dir in sub_dirs:
-            (self.source / sub_dir).mkdir(parents=True, exist_ok=True)
+            (source / sub_dir).mkdir(parents=True, exist_ok=True)
 
     def read(self, unique_id: str) -> str:
         """reads data corresponding to identifier"""
-        # before open should make absolute path
-        with open_(self.source / unique_id) as input:
-            data = input.read()
-            if self._md5 and isinstance(data, str):
-                self._checksums[unique_id] = get_text_hexdigest(data)
+        with open_(self.source / unique_id) as infile:
+            data = infile.read()
 
```

Comment:
I think you have removed `checksum` and `get_text_hexdigest`. Without `get_text_hexdigest`, how we have md5?

Reply:
The `md5` is calculated and written on the write operation. This is in the `_write()` method under the `md5` clause. ([See here](https://github.com/cogent3/cogent3/pull/1169/files#diff-ae10de901aabb1e7506cd860c9fd7b0d9b464b7ccd6285e25df18d00d611325dL324).)